packages/middleware-bucket-endpoint/src/bucketHostname.spec.ts

```diff
@@ -713,4 +713,210 @@ describe("bucketHostname", () => {
       expect(signingService).toBe("s3-outposts");
     });
   });
+
+  describe("from Object Lamdba ARN", () => {
+    describe("populates access point endpoint from ARN", () => {
+      it("should use the proper endpoint", () => {
+        const region = "eu-west-1";
+        const expectedEndpoint = "js-sdk-ap-name-123456789012.s3-object-lambda.eu-west-1.amazonaws.com";
+        ["arn:aws:s3-object-lambda:eu-west-1:123456789012:accesspoint/js-sdk-ap-name"].forEach((outpostArn) => {
+          const { bucketEndpoint, hostname } = bucketHostname({
+            bucketName: parseArn(outpostArn),
+            baseHostname: "s3.eu-west-1.amazonaws.com",
+            isCustomEndpoint: false,
+            clientRegion: region,
+          });
+          expect(bucketEndpoint).toBe(true);
+          expect(hostname).toBe(expectedEndpoint);
+        });
+      });
+    });
+
+    it("should not be able to use accelerate", () => {
+      try {
+        bucketHostname({
+          bucketName: parseArn("arn:aws:s3-object-lambda:us-west-2:123456789012:accesspoint/mybanner"),
+          baseHostname: "s3.us-west-2.amazonaws.com",
+          isCustomEndpoint: false,
+          clientRegion: "us-west-2",
+          accelerateEndpoint: true,
+        });
+        // should never get here
+        expect(1).toEqual(0);
```

Comment:
Nit: this won't provide very useful error message, you can validate exactly 1 assertion is called by `expect.assertions(1);`. Same to other unit tests.

Reply:
ahh TIL. I'll update

Reply:
Not entirely sure how to replace this with `expect.assertion()` since this expect shouldn't ever be called here to begin with. I'm thinking possibly change the error message? Maybe something like:

```
expect("This configuration is invalid and should never get to this line").toEqual(0);
```